File: server/service/domain_list.go
package service

import (
	"gin-vue-admin/global"
	"gin-vue-admin/model"
	"gin-vue-admin/model/request"
	"gin-vue-admin/service/ali"
	"gin-vue-admin/service/dnspod"
	"strings"
)

// CreateDomain 创建Domain记录
// Author [dukanghub](https://github.com/Dukanghub)
func CreateDomain(domain model.DomainInfo) (err error) {
	err = global.GVA_DB.Create(&domain).Error
	return err
}

// DeleteDomain 删除Domain记录
// Author [dukanghub](https://github.com/Dukanghub)
func DeleteDomain(domain model.DomainInfo) (err error) {
	err = global.GVA_DB.Delete(&domain).Error
	return err
}

// DeleteDomainByIds 批量删除Domain记录
// Author [dukanghub](https://github.com/Dukanghub)
func DeleteDomainByIds(ids request.IdsReq) (err error) {
	err = global.GVA_DB.Delete(&[]model.DomainInfo{},"id in ?",ids.Ids).Error
	return err
}

// UpdateDomain 更新Domain记录
// Author [dukanghub](https://github.com/Dukanghub)
func UpdateDomain(domain model.DomainInfo) (err error) {
	err = global.GVA_DB.Save(&domain).Error
	return err
}

// GetDomainById 根据id获取Domain记录
// Author [dukanghub](https://github.com/Dukanghub)
func GetDomainById(id uint) (err error, domain model.DomainInfo) {
	err = global.GVA_DB.Where("id = ?", id).First(&domain).Error
	return
}

// GetDomainList 分页获取Domain记录
// Author [dukanghub](https://github.com/Dukanghub)
func GetDomainList(info request.DomainListSearch) (err error, list interface{}, total int64) {
	limit := info.PageSize
	offset := info.PageSize * (info.Page - 1)
	//fmt.Println("info.DnsProvider", info.DnsProvider)
    // 创建db
	db := global.GVA_DB.Model(&model.DomainInfo{})
    var domains []model.DomainInfo
    // 如果有条件搜索 下方会自动创建搜索语句
    if info.RecordTotal != 0 {
        db = db.Where("`record_total` > ?",info.RecordTotal)
    }
    if info.DnsProvider != 0 {
        db = db.Where("`dns_provider` = ?",info.DnsProvider)
    }
	if info.Company != 0 {
		db = db.Where("`company` = ?",info.Company)
	}
    if info.Account != 0 {
        db = db.Where("`account` = ?",info.Account)
    }
	if strings.Contains(info.Domain, ",") {
		domainArr := strings.Split(info.Domain, ",")
		db = db.Where("`domain` in (?)", domainArr)
	} else if info.Domain != "" {
		db = db.Where("`domain` = ?",info.Domain)
	}
	db = db.Order("record_total desc").Order("updated_at desc").Order("company")
	err = db.Count(&total).Error
	err = db.Limit(limit).Offset(offset).Find(&domains).Error
	return err, domains, total
}

// UpdateDomains 批量更新Domain记录
// Author [dukanghub](https://github.com/Dukanghub)
func UpdateDomains(domains model.Domains) (err error) {

	for _,v := range domains.Domains {

		results := global.GVA_DB.Where("domain_id = ?", v.DomainId).Updates(&v)
		if results.Error != nil {
			err = results.Error
		}
	}
	return err
}

// FlushDomainsToDb 更新账号下的域名列表到数据库
// Author [dukanghub](https://github.com/Dukanghub)
func FlushDomainsToDb() (err error) {
	var users []model.DnsUser
	err = global.GVA_DB.Find(&users).Error
	if err != nil {
		return err
	}
	for _,v := range users {
		if v.Platform == 1 {
			err = ali.FlushDomainsToDb(v)
			//go ali.FlushDomainsToDb(v)
			if err != nil {
				return err
			}
		} else if v.Platform == 2 {
			err = dnspod.FlushDomainsToDb(v)
			if err != nil {
				return err
			}
		}
	}
	return err
}

File: web/src/api/domainList.js
import service from '@/utils/request'

// @Tags DomainInfo
// @Summary 创建Domain
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DomainInfo true "创建Domain"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /domainList/createDomain [post]
export const createDomain = (data) => {
  return service({
    url: '/domainList/createDomain',
    method: 'post',
    data
  })
}

// DeleteDomain 删除Domain
// @Tags DomainInfo
// @Summary 删除Domain
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DomainInfo true "删除Domain"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /domainList/deleteDomain [delete]
export const deleteDomain = (data) => {
  return service({
    url: '/domainList/deleteDomain',
    method: 'delete',
    data
  })
}

// @Tags DomainInfo
// @Summary 批量删除Domains
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除Domains"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"批量删除成功"}"
// @Router /domainList/deleteDomainByIds [delete]
export const deleteDomainByIds = (data) => {
  return service({
    url: '/domainList/deleteDomainByIds',
    method: 'delete',
    data
  })
}

// @Tags DomainInfo
// @Summary 更新Domain
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DomainInfo true "更新Domain"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /domainList/updateDomain [put]
export const updateDomain = (data) => {
  return service({
    url: '/domainList/updateDomain',
    method: 'put',
    data
  })
}

// @Tags DomainInfo
// @Summary 批量更新Domain
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body []model.DomainInfo true "批量更新Domain"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"批量更新成功"}"
// @Router /domainList/updateDomains [put]
export const updateDomains = (data) => {
  return service({
    url: '/domainList/updateDomains',
    method: 'put',
    data
  })
}

// @Tags DomainInfo
// @Summary 用id查询Domain
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DomainInfo true "用id查询Domain"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /domainList/findDomain [get]
export const findDomain = (params) => {
  return service({
    url: '/domainList/findDomain',
    method: 'get',
    params
  })
}

// @Tags DomainInfo
// @Summary 分页获取Domain列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.DomainListSearch true "分页获取Domain列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /domainList/getDomainList [get]
export const getDomainList = (params) => {
  return service({
    url: '/domainList/getDomainList',
    method: 'get',
    params
  })
}

// @Tags DomainInfo
// @Summary 从线上获取最新域名信息并更新到数据库
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /domainList/flushDomains [get]
export const flushDomains = (params) => {
  return service({
    url: '/domainList/flushDomains',
    method: 'get',
    params
  })
}
